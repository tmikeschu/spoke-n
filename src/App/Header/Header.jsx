import React from 'react'
import {  Link } from 'react-router'
import logo from './wheel.svg'
import '../App.css'

const Header = () => (
  <article className="header">
    <img src={logo} className="header-logo" alt="logo" />
    <h3><Link to="/landing">The Spoken Tour</Link></h3>
  </article>
)

export default Header
