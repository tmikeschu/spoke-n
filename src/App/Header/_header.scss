.Header {
  display: flex;
  align-items: center;
  background-color: $light;
  h3 {
    a {
      color: $dark;
    }
  }
  &__logo {
    width: 40px;
    margin-right: 16px;
  }

  @media (max-width: 640px) {
    &__logo {
      display: none;
    }
  }
}
