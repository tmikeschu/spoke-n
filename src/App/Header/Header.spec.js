import React, { Component } from "react"
import { shallow } from "enzyme"
import Header from "./Header"
import { Link } from 'react-router'
import logo from './wheel.svg'

describe("<Header />", () => {
  const header = shallow(<Header />)

  it("renders without crashing", () => {
    expect(header).toBeTruthy()
  })

  it("renders the correct content", () => {
    expect(header.equals(
      <article className="header">
        <img src={logo} className="header-logo" alt="logo" />
        <h3><Link to="/landing">The Spoken Tour</Link></h3>
      </article>
    )).toBeTruthy()
  })

  describe("snapshot", () => {
    it("is valid", () => {
      expect(header.getNodes()).toMatchSnapshot()
    })
  })
})

