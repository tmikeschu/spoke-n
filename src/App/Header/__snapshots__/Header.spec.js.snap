// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> snapshot is valid 1`] = `
Array [
  <article
    className="header"
  >
    <img
      alt="logo"
      className="header-logo"
      src="wheel.svg"
    />
    <h3>
      <Link
        onlyActiveOnIndex={false}
        style={Object {}}
        to="/landing"
      >
        The Spoken Tour
      </Link>
    </h3>
  </article>,
]
`;
